frontend/demo/src/reducers/data/messages/index.ts

```diff
@@ -0,0 +1,46 @@
+import {ActionType, getType} from 'typesafe-actions';
+
+import * as actions from '../../../actions/messages';
+import {Message} from 'httpclient';
+import {DataState} from '..';
+
+type Action = ActionType<typeof actions>;
+
+export type MessagesState = {
+  data: DataState;
+};
+
+export type MessageMap = {
+  [messageId: string]: Message;
+};
+
+export type Messages = {
+  all: {[conversationId: string]: MessageMap};
+};
+
+const initialState = {
+  all: {},
+};
+
+function organiseMessages(messages: Message[]): MessageMap {
+  const newMessages: MessageMap = {};
+  messages.forEach((message: Message) => {
+    newMessages[message.id] = message;
+  });
+  return newMessages;
```

Comment:
`_.keyBy(messages)` should do the same (we already have lodash right?)

Reply:
`return _.keyBy(messages, 'id');`